docs/metadata.rst

```diff
@@ -107,6 +107,9 @@ explanatory purposes.
         userdoc: [ ... ],
         // Required: NatSpec developer documentation of the contract
         devdoc: [ ... ],
+        // Optional: Information whether the contract is a library or not.
```

Comment:
In what use cases is this needed?

Reply:
In burrow we match the code of an account against the DeployedBytecode from the solidity compiler so we can say which contract it is (makes it possible to store metadata/abis on-chain and match them against newly created contracts).

However for libraries DeployedBytecode output does not match the actual deployed code due to https://solidity.readthedocs.io/en/v0.5.11/contracts.html?highlight=call%20protection#call-protection-for-libraries

So in burrow we work around this like so:
https://github.com/hyperledger/burrow/blob/develop/deploy/compile/compilers.go#L459
This code is not as ugly as the initial version of this code, so I don't think it is as important as I originally thought.

Reply:
Well if you need checks done in the runtime bytecode, adding this field into the metadata JSON doesn't seem to be all that useful. However adding it into the metadata CBOR could be of use.

Reply:
In this case we're not using the cbor metadata. We're storing evm.DeployedByte.Object (amongst others) so we can at runtime answer the question "what is the abi/contractname/solidity source/etc"  for the code of this account. (Note it would be nice to have library or not in the output of burrow accounts, see https://github.com/hyperledger/burrow/commit/5bb22a515475eca57a84c9b6613ba2742376b0f2).

So extracting this from the json output rather than the metadata makes no difference to burrow.


Having said all of that, I think the question is where does this belong. Is it general compiler output or metadata information? Maybe metadata does seem like a better place.